Add AWS Exporter to the docs (#211)
port-labs/port-docs: docs/build-your-software-catalog/sync-data-to-catalog/aws/aws.md

- Use relations to create a complete, easily digestible map of your AWS accounts inside Port;
- etc.

## How it works

Port's AWS exporter can retrieve all the resources supported by the [AWS Cloud Control API](https://docs.aws.amazon.com/cloudcontrolapi/latest/userguide/what-is-cloudcontrolapi.html).

The open source AWS exporter allows you to perform extract, transform, load (ETL) on data from AWS into the desired software catalog data model.

The exporter is deployed using an [AWS serverless application](https://aws.amazon.com/serverless/sam/) that is installed on the account.

[The serverless application](#exporter-aws-serverless-application) requires a [JSON configuration file](#exporter-configjson-file) to describe the ETL process to load data into the developer portal, and an [IAM policy](#iam-policy) with the necessary permissions to list and read the configured resources.

The exporter makes use of [JQ JSON processor](https://stedolan.github.io/jq/manual/) to select, modify, concatenate, transform and perform other operations on existing fields and values from the AWS objects.

### Exporter `config.json` file

> The `config.json` file is how you specify the exact resources you want to export from your AWS account, and also which entities and which properties in Port, you want to fill in with data.

Reviewer: ```suggestion
The `config.json` file is how you specify the exact resources you want to export from your AWS account, and also which Entities and which properties in Port, you want to fill in with data.
```

  :::

  Some example use cases:

  - To sync all objects from the specified `kind` in the default region - do not specify a `selector`;
  - To sync all lambdas that are not owned by AWS Amplify service:

    ```json showLineNumbers
    query: .FunctionName | startswith("amplify") | not
    ```

  - etc.

  :::info

  - For some resources you have to provide additional information. For example, in order to sync all `AWS ELB Listeners` of a specific load balancer, use the `regions_config` and `resources_models` keys:

    ```json showLineNumbers
    "selector": {
        "aws": {
          "regions": [
            "eu-west-1"
          ],
          # highlight-start
          "regions_config": {
            "eu-west-1": {
              "resources_models": [
                "{\"LoadBalancerArn\": \"<AWS_ELB_ARN>\"}"
              ]
            }
          }
          # highlight-end
        }
    }
    ```

  The table of the special resources with the required properties can be found [here](https://docs.aws.amazon.com/cloudcontrolapi/latest/userguide/resource-operations-list.html#resource-operations-list-containers).
  :::

> - The `port`, `entity` and the `mappings` keys open the section used to map the AWS resource fields to Port entities, the `mappings` key is an array where each object matches the structure of an [entity](../sync-data-to-catalog.md#entity-json-structure).

Reviewer: ```suggestion
- The `port`, `entity` and the `mappings` keys open the section used to map the AWS resource fields to Port Entities, the `mappings` key is an array where each object matches the structure of an [entity](../sync-data-to-catalog.md#entity-json-structure).
```

              "title": ".FunctionName",
              "blueprint": "function",
              "properties": {
                "memory": ".MemorySize",
                "timeout": ".Timeout",
                "runtime": ".Runtime"
              }
            }
          ]
        }
      }
      # highlight-end
    }
    ...
  ```

  :::info **IMPORTANT**

  - **The order of the resources matters** when you have relations between resources.
    The AWS exporter will sync the resources in the same order as they appear in the `config.json`, so make sure to sort the resources by a logical order.

    For example, if you have a relation from SNS Topic to lambda function, put the Lambda function configuration first.

  - By its nature, the AWS exporter will keep the values of unmapped properties untouched. Go [here](../api/api.md?operation=create-update#usage) for further explanation about different entity creation strategies.

  :::

  :::tip View a resource type schema
  To view a resource type schema and use it to compose a mapping, use [this](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-template-resource-type-ref.html) reference.

>   Pay attention that not all of the resource types listed in the reference are available for use with the Cloud Control API. A method to determine if a resource type is available discussed [here](#structure).

Reviewer: ```suggestion
  Note that not all of the resource types listed in the reference are available for use with the Cloud Control API. A method to determine if a resource type is available discussed [here](#structure).
```

      "Resource": "*"
    }
  ]
}
```

:::tip Find resource actions for policy
In order to find the resource actions that should be added to the policy, you should view the resource type schema, and locate the permissions for the **read** and **list** `handlers`.

Use the following command to get the resource action for a specific `<RESOURCE_TYPE>` (AWS CLI and JQ required):

```bash showLineNumbers
aws cloudformation describe-type --type RESOURCE --type-name <RESOURCE_TYPE> --query "Schema" | jq -c 'fromjson | .handlers | with_entries(select([.key] | inside(["list", "read"]))) | map(.permissions) | flatten'
```

More details can be found [here](https://docs.aws.amazon.com/cloudcontrolapi/latest/userguide/resource-types.html#resource-types-schemas).
:::

### Exporter S3 Bucket

The exporter's [`config.json`](#exporter-configjson-file) file needs to be saved to an [`AWS S3 Bucket`](https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingBucket.html).

You can create and manage your own bucket or let the exporter do that for you.
In any case, you need to upload the [`config.json`](#exporter-configjson-file) to the bucket, when it's available.

### Port Credentials Secret

> In order to manage entities in Port, the exporter needs access to your Port credentials, these are provided to the exporter via an [`AWS Secrets Manager`](https://docs.aws.amazon.com/secretsmanager/latest/userguide/intro.html) secret.

Reviewer: ```suggestion
In order to manage Entities in Port, the exporter needs access to your Port credentials, these are provided to the exporter via an [`AWS Secrets Manager`](https://docs.aws.amazon.com/secretsmanager/latest/userguide/intro.html) secret.
```
